SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -97,69 +96,78 @@ void USpatialTypebindingManager::CreateTypebindings()
 				}
 			}
 		}
-		
-		Info.SingleClientComponent = SchemaDatabase->ClassToSchema[Class].SingleClientRepData;
-		ComponentToClassMap.Add(Info.SingleClientComponent, Class);
 
-		Info.MultiClientComponent = SchemaDatabase->ClassToSchema[Class].MultiClientRepData;
-		ComponentToClassMap.Add(Info.MultiClientComponent, Class);
-
-		Info.HandoverComponent = SchemaDatabase->ClassToSchema[Class].HandoverData;
-		ComponentToClassMap.Add(Info.HandoverComponent, Class);
+		ForAllSchemaComponentTypes([&](EComponentType Type) {
+			Worker_ComponentId ComponentId = SchemaDatabase->ClassToSchema[Class].SchemaComponents[Type];
+			if (ComponentId != 0)
+			{
+				Info.SchemaComponents[Type] = ComponentId;
+				ComponentToClassMap.Add(ComponentId, Class);
+				ComponentToOffsetMap.Add(ComponentId, 0);
+				ComponentToCategoryMap.Add(ComponentId, (EComponentType)Type);
+			}
+		});
 
-		Info.RPCComponents[RPC_Client] = SchemaDatabase->ClassToSchema[Class].ClientRPCs;
-		ComponentToClassMap.Add(Info.RPCComponents[RPC_Client], Class);
+		Info.Class = Class;
 
-		Info.RPCComponents[RPC_Server] = SchemaDatabase->ClassToSchema[Class].ServerRPCs;
-		ComponentToClassMap.Add(Info.RPCComponents[RPC_Server], Class);
+		ClassInfoMap.Add(Class, MakeShared<FClassInfo>(Info));
+	}
 
-		Info.RPCComponents[RPC_NetMulticast] = SchemaDatabase->ClassToSchema[Class].NetMulticastRPCs;
-		ComponentToClassMap.Add(Info.RPCComponents[RPC_NetMulticast], Class);
+	for (UClass* Class : SupportedClasses)
+	{
+		for (auto& SubobjectDataPair : SchemaDatabase->ClassToSchema[Class].SubobjectData)
+		{
+			int32 Offset = SubobjectDataPair.Key;
+			FSubobjectSchemaData SubobjectSchemaData = SubobjectDataPair.Value;
 
-		Info.RPCComponents[RPC_CrossServer] = SchemaDatabase->ClassToSchema[Class].CrossServerRPCs;
-		ComponentToClassMap.Add(Info.RPCComponents[RPC_CrossServer], Class);
+			FClassInfo* ActorInfo = FindClassInfoByClass(Class);
+			FClassInfo SubobjectInfo = *FindClassInfoByClass(SubobjectSchemaData.Class);
```

Comment:
Don't like this dereference as FindClass could return null.

Reply:
Yeah, we should probably log Fatal saying the SchemaDatabase is malformed and needs to be regenerated.